source/mainnet/tutorials/using-ID-in-dApps/zk-proofs-tutorial.rst

```diff
@@ -0,0 +1,299 @@
+.. _zk-proofs-tutorial:
+
+===================
+ZK proofs structure
+===================
+
+Now we'll explore how to implement zero-knowledge proof verification using Concordium's identity framework, focusing on how ZK statements are structured.
+
+Defining ZK statements
+----------------------
+
+First, let's look at the ZK statements defined in the constants file:
+
+.. code-block:: javascript
+
+  import { AtomicStatementV2 } from "@concordium/web-sdk"; // From constants.ts
+
+  // Context string for signing messages and generating ZK proofs
+  export const CONTEXT_STRING = "CONCORDIUM_COMPLIANT_REWARD_DISTRIBUTION_DAPP";
+
+  // ZK statements for the verifiable presentation
+  export const ZK_STATEMENTS = [
+      {
+      type: "RevealAttribute",
+      attributeTag: "nationalIdNo",
+      },
+      {
+      type: "RevealAttribute",
+      attributeTag: "nationality",
+      },
+      {
+      type: "AttributeInRange",
+      attributeTag: "dob",
+      lower: "18000101",
+      upper: "20060802",  // Ensures user is over 18 years old
+      },
+      {
+      type: "AttributeNotInSet",
+      attributeTag: "countryOfResidence",
+      set: ["US", "KP", "RU"],  // Not from USA, North Korea, or Russia
+      },
+  ] as AtomicStatementV2[];
+
+  // The number of blocks after the `best block` (top of chain), where the `recent block` is located.
+  // The `recent block hash` is included in signatures and ZK proofs to ensure they expire.
```

Comment:
Worth to mention that the `verifier` needs to check that a recent `block_hash` was used in the ZK proof for proofs to expiry.